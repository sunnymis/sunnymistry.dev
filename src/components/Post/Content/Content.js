import React from 'react';
import moment from 'moment';
import styles from './Content.module.scss';

const Content = ({ body, title, date }) => (
  <div className={styles['content']}>
    <h1 className={styles['content__title']}>{title}</h1>
    <p className={styles['content__date']}>{moment(date).format('MMMM D, YYYY')}</p>
    <div className={styles['content__body']} dangerouslySetInnerHTML={{ __html: body }} />
  </div>
);

export default Content;
