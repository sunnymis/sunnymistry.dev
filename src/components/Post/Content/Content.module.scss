@import '../../../assets/scss/variables';
@import '../../../assets/scss/mixins';

.content {
  max-width: $layout-post-single-width;
  padding: 0 15px;
  margin: 0 auto;

  &__date {
    font-style: italic;
    color: $color-gray;
    text-align: center;
  }
  &__title {
    font-size: $typographic-base-font-size * 2;
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    text-align: center;
    @include line-height(1.65);
    @include margin-top(1);
    @include margin-bottom(0)
  }

  &__body {
    & figure {
      @include margin-bottom(1);

      & blockquote {
        font-style: italic;
        text-align: center;
        margin-top: 0;
        @include padding(1, 0);

        & p {
          max-width: $layout-post-width;
          font-size: $typographic-base-font-size * 1.6817;
          margin-top: 0;
          @include margin-bottom(1);
          @include line-height(1.5)
        }

      }

    }

    & * {
      max-width: $layout-post-width;
      margin-left: auto;
      margin-right: auto
    }

    & img {
      max-width: 100%;
    }

    ul {
      li {
        margin-bottom: 12px;
        
        p {
          margin: 5px 0;
        }
      }
    }

    // ul {
    //   li {
    //     // &:before {
    //     //   content: '';
    //     //   display: inline-block;
    //     //   position: relative;
    //     //   top: -2px;
    //     //   height: 7px;
    //     //   width: 7px;
    //     //   margin-right: 1em;
    //     //   margin-left: -1em;
    //     //   background-color: $color-primary;
    //     //   border-radius: 50px;
    //     // }
    //   }
    // }

    ul, p {
      & a {
        text-decoration: none;
        color: $color-base;
        padding-bottom: 3px;
        border-bottom: 1px solid $color-primary;

        &:hover {
          color: $color-primary;
        }
      }
    }

    code {
      white-space: pre-wrap;
    }

  }

}

@include breakpoint-md {
  .content {
    padding: 0;

    &__title {
      font-size: $typographic-base-font-size * 3;
      @include line-height(2.25);
      @include margin-top(2.25);
      @include margin-bottom(1)
    }

    &__body {
      font-size: $typographic-base-font-size * 1.125;
      @include line-height(1.125);
      @include margin-bottom(1.125);

      & p {
        font-size: $typographic-base-font-size * 1.125;
        @include line-height(1.125);
        @include margin-bottom(1.125)
      }

    }

  }

}
