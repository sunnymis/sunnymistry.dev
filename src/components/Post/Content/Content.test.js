import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';
import Content from './Content';

jest.mock('moment', () => {
  const mockedMoment = {
    format: jest.fn(),
  };
  return jest.fn(() => mockedMoment);
});

describe('Content', () => {
  it('renders correctly', () => {
    moment().format.mockReturnValueOnce('December 26, 2019');

    const props = {
      title: 'test',
      body: '<p>test</p>'
    };

    const tree = renderer.create(<Content {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
