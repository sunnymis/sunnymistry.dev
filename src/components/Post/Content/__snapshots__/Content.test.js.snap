// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="content"
>
  <h1
    className="content__title"
  >
    test
  </h1>
  <p
    className="content__date"
  >
    December 26, 2019
  </p>
  <div
    className="content__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
